### ISSUE TITLE
Screensaver turns on in linux using gamepad

### ISSUE THREAD

**Issue by rosskidson** [OneLoneCoder/olcPixelGameEngine#207]
In linux, if one only provides inputs via a gamepad using the gamepad extension, the screensaver is not blocked and eventually comes on.  This doesn't seem to be a problem in Windows, and the gamepad extension does not support MacOS, so this just seems to be a Linux problem.

My current workaround is to add the following to `Platform_Linux::HandleSystemEvent()`:
```
XResetScreenSaver(olc_Display);
```

Should I set up a pull request for this?  Or is there a better way to solve the problem?

**Comment by OneLoneCoder** [comment]
Hello!
Good find! The easiest way is probably to raise the issue with developer of the game pad extension (which im assuming is gorbit99) https://github.com/gorbit99/olcPGEX_Gamepad ? I'll also drop him a note on the discord.

**Comment by rosskidson** [comment]
Thanks for the quick response!

Yeah so that was also a thought I had, however the `olc::GamePad` class is a standalone class that doesn't have any access to  `olc::PixelGameEngine` internals, which is needs to gain access to `olcDisplay`.  However having said that it could inherit from `PGEX` in order to gain access, so I think that would probably be the best way of going about this.

**Comment by gorbit99** [comment]
HI!
I'd be happy if you could raise an issue on the gamepad extension repo
https://github.com/gorbit99/olcPGEX_Gamepad
The issue is, that friendness can't be inherited last ai checked, so it simply cannot access internals, but I'll see what I can do.

**Comment by OneLoneCoder** [comment]
Looks like we'll need a joint effort on this. I'll see what I can put into PGE 2.10 to expose a few more internals.

**Comment by rosskidson** [comment]
Indeed.  It seems that the `X11::Display* olc_Display` is a private of `Platform_Linux`, which is stored in a static global unique pointer.

I got a working version by adding the following to `Platform_Linux`:
```
X11::Display* GetX11Display() {
  return olc_Display;
}
```

And then to `olc::GamePad::poll()`:
```
auto linux_platform = dynamic_cast<Platform_Linux*>(platform.get());                                                                                                
if(linux_platform && linux_platform->GetX11Display()) { 
  XResetScreenSaver(linux_platform->GetX11Display());
}                                                                                                    
```

Still just a proof of concept, I'm not sure that a dynamic_cast on every poll is ideal.

**Comment by gorbit99** [comment]
Maybe have something like a few getters for certain internals inside the pgex, if it's a friend already anyways, that way you don't have random bits and bobs visible to the regular user, but a pgex could use them

**Comment by gorbit99** [comment]
@rosskidson Can you please try this version:
[gamepad.zip](https://github.com/OneLoneCoder/olcPixelGameEngine/files/5836127/gamepad.zip)
It will be very much a breaking update, but it should also solve the screensaver issue.

**Comment by rosskidson** [comment]
@gorbit99 Sorry for the long delay!

I tried it and this worked out for me.  I think I'll stick with my work around until a change is integrated.

**Comment by gorbit99** [comment]
@rosskidson I did push those already
